### PR TITLE
Add function to return Nth instance of substring

### PR DESCRIPTION
**Pull request by codope**
Implements [issue 10771](https://github.com/prestodb/presto/issues/10771)

### CODE REVIEW

File: presto-docs/src/main/sphinx/functions/teradata.rst [hunk 1]
```diff
@@ -11,7 +11,7 @@ String Functions
 
     Returns the hexadecimal representation of the UTF-16BE encoding of the string.
 
-.. function:: index(string, substring) -> bigint
+.. function:: index(string, substring, instance) -> bigint
```
Comment (findepi) [inline presto-docs/src/main/sphinx/functions/teradata.rst]: see https://www.info.teradata.com/HTMLPubs/DB_TTU_16_00/SQL_Reference/B035-1145-160K/gcn1472241145389.html#wwID0ENTAM

Comment (codope) [reply presto-docs/src/main/sphinx/functions/teradata.rst]: @findepi thanks for the doc. In that case, I will have to decouple Teradata's index() from strpos(). Is there a way to make an argument of strpos() optional? Or, should I implement a new method altogether?

Comment (codope) [reply presto-docs/src/main/sphinx/functions/teradata.rst]: I have undone the changes related to Terdata index() function and added a separate stringPosition() method that takes instance as an argument. Please review the changes.

**Comment by findepi** [comment]
@electrum  https://github.com/prestodb/presto/issues/10771 is still marked `syntax-needs-review`

**Comment by codope** [comment]
@findepi  @electrum 
Could you please review this ticket?

**Comment by findepi** [comment]
@codope:
> #10771 is still marked `syntax-needs-review`

i will NOT be reviewing this.

Also, for anyone to be able to review this, you would need to remove `Merge remote-tracking branch 'upstream/master'...` and `undo...` commits (generally rebase on master and squash all commits into one).

**Comment by codope** [comment]
@rongrong @findepi Could you please decline this PR? I have rebased my changes on top of master and checked out a different branch. Here's the new PR: [12664](https://github.com/prestodb/presto/pull/12664)

**Comment by rongrong** [comment]
In favor of #12664

**Pull request closed without merging**